packages/types/src/checkin.ts

```diff
@@ -38,16 +38,25 @@ export interface SerializedCheckIn {
   };
 }
 
-export type CheckIn = {
+interface InProgressCheckIn {
   // The distinct slug of the monitor.
   monitorSlug: SerializedCheckIn['monitor_slug'];
+  // The status of the check-in.
+  status: 'in_progress';
+}
+
+interface FinishedCheckIn {
+  // The distinct slug of the monitor.
+  monitorSlug: SerializedCheckIn['monitor_slug'];
+  // The status of the check-in.
+  status: 'ok' | 'error';
   // Check-In ID (unique and client generated).
   checkInId?: SerializedCheckIn['check_in_id'];
```

Comment:
Do we want to make the ID required for FinishedCheckIns?

Reply:
Honestly, why the hell do we need both an ID and a name??

Reply:
> Do we want to make the ID required for FinishedCheckIns

It makes the DX worse, so no. This is because technically an ID can be undefined if the client is disabled (so the first `captureCheckIn` will return undefined in that case)

> Honestly, why the hell do we need both an ID and a name??

It's because you can have multiple crons with the same name fire at the same time, the id helps make sure that this works.

Reply:
Just to explain the point I was making here because I think you misunderstood me here: For the finished checkins we theoretically don't need to send the id AND the slug since we can already map from id to slug in the product.

I still stand by my point that we should not be required to send both but I'll approve this to unblock so that our current implementation is not busted.